tapdb/assets_store.go

```diff
@@ -1477,6 +1477,10 @@ func locatorToProofQuery(locator proof.Locator) (FetchAssetProof, error) {
 		args.Outpoint = outpoint
 	}
 
+	if locator.AssetID != nil {
+		args.AssetID = locator.AssetID[:]
```

Comment:
Hmm, yeh looks like another instance where we forgot to handle/prevent the collision case that can happen here. 

Can we add a test case here to make sure we can properly select based on the final 3-tuple (assetID, scriptKey, outpoint)?

Reply:
Added a test case.